libs/sr/ui/src/LightCone/LightConeEditor/LightConeEditor.tsx

```diff
@@ -0,0 +1,255 @@
+
+// TODO: temporary until light cone sheet is implemented
+interface ILightConeSheet {
+  superimpose: readonly SuperimposeKey[]
+}
+
+const tempLightConeSheet: ILightConeSheet = {
+  superimpose: [1, 2, 3, 4, 5],
+}
+
+// TODO: light cone sheets, errors, autocomplete, display text, i18n, ...
+export type LightConeEditorProps = {
+  lightConeIdToEdit?: string
+}
+
+export function LightConeEditor({
+  lightConeIdToEdit = 'new',
+}: LightConeEditorProps) {
+  const { t } = useTranslation('lightCone')
+  const { database } = useDatabaseContext()
+  const [dirtyDatabase, setDirtyDatabase] = useForceUpdate()
+
+  useEffect(
+    () => database.lightCones.followAny(setDirtyDatabase),
+    [database, setDirtyDatabase]
+  )
+
+  useEffect(() => {
+    if (lightConeIdToEdit === 'new') {
+      dispatchLightCone({ type: 'reset' })
+    }
+    const dbLightCone =
+      lightConeIdToEdit &&
+      dirtyDatabase &&
+      database.lightCones.get(lightConeIdToEdit)
+    if (dbLightCone) {
+      dispatchLightCone({
+        type: 'overwrite',
+        lightCone: deepClone(dbLightCone),
+      })
+    }
+  }, [lightConeIdToEdit, database, dirtyDatabase])
+
+  const [lightCone, dispatchLightCone] = useReducer(lightConeReducer, undefined)
+  const sheet: ILightConeSheet | undefined = lightCone
+    ? tempLightConeSheet
+    : undefined
+
+  const update = useCallback(
+    (newValue: Partial<ILightCone>) => {
+      // const newSheet = newValue.key ? getLightConeSheet(newValue.key) : sheet!
+      const newSheet = newValue.key ? tempLightConeSheet : sheet!
+
+      function pick<T>(
+        value: T | undefined,
+        available: readonly T[],
+        prefer?: T
+      ): T {
+        return value && available.includes(value)
+          ? value
+          : prefer ?? available[0]
+      }
+
+      if (newValue.key) {
+        newValue.superimpose = pick(
+          lightCone?.superimpose,
+          newSheet.superimpose,
+          Math.min(...newSheet.superimpose) as SuperimposeKey
+        )
+
+        // for some reason, location property not being added (undefined) if no character location is set, so force init
+        newValue.location = newValue.location ?? ''
+      }
+      if (newValue.level) {
+        newValue.level = clamp(newValue.level, 0, lightConeMaxLevel)
+      }
+
+      dispatchLightCone({ type: 'update', lightCone: newValue })
+    },
+    [lightCone, sheet, dispatchLightCone]
+  )
+
+  const reset = useCallback(() => {
+    dispatchLightCone({ type: 'reset' })
+  }, [])
+
+  useEffect(() => {
+    if (lightConeIdToEdit === 'new') {
+      dispatchLightCone({ type: 'reset' })
+    }
+  }, [lightConeIdToEdit])
+
+  return (
+    <Container>
+      <CardThemed bgt="dark">
+        <CardHeader title="Light Cone Editor" />
+        <CardContent sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}>
+          <Grid container spacing={1} columns={{ xs: 1, md: 2 }}>
+            {/* name */}
```

Comment:
I think it will be cleaner if we separate each form input into a separate component. Also if there is any overlap between level dropdown for LC and char like you mentioned, it will be easier to split out.

The component can stay inside this file, no need to export it, for now

Reply:
Correct